### DISCUSSION TITLE
Request: Volume Control of Non-Standard Device

### DISCUSSION THREAD

**Discussion by darkman1984** [File-New-Project/EarTrumpet#671] [Ideas]
Hey there!

Is it possible that you implement the option to set which device (also not the standard device) is controlable by the volume keys of a media-keyboard or by mouse (volume scrollwheel)? Would be very nice.

Greetings
Chris

**Comment by CortexReaver** [comment]
Bump, seems like this is still not being implemented. Right now EarTrumpet allows to switch only between the default audio outputs. It's fast and cool, no need to go deep into windows settings, but the thing is when you have a bunch of virtual cables there is an urgency to volume control the whole speakers and not an default audio input.
All in all you ending up with installing a dozen of different programs, doing crazy workarounds, just for one simple task (including EarTrumpet itself), like it was described here: https://superuser.com/questions/700849/change-device-controlled-by-volume-keys-without-changing-default

**Reply by riverar** [reply]
Check out EarTrumpet Actions in the dev builds. Should be able to control devices regardless of if they are default or not.

**Reply by CortexReaver** [reply]
I've downloaded this: https://github.com/File-New-Project/EarTrumpet/actions/runs/5328756519 (unpacked "loose.zip" in a corresponding folder by hand) This is the main branch, right? Cannot find where I can switch media keys controls anyway. Right click context menu switches default device as before.

**Reply by CortexReaver** [reply]
Figured this out just by hooking "Increase\decrease volume for all devices" to the media keys in settings. But it seems like there is no indication of volume levels on screen anymore :\ And I really got used to this Win 10 feature.

**Reply by CortexReaver** [reply]
Played with it a little, and I think EarTrumpet need also an option to set "mute all devices" hotkey. Because right now mute media key mutes only default device (virtual cable).

**Reply by riverar** [reply]
Hey @CortexReaver, our dev builds feed from the `dev` branch and can be installed via https://install.eartrumpet.app/dev/EarTrumpet.Package.appinstaller. Try that out!

**Reply by CortexReaver** [reply]
EarTrumpet was already installed with Chocolatey early on my machine, but I don't know shell commands for installing nightly builds/different branches. And there is also something called "appinstaller" you are telling me about right now. Do I need to install this too? This is all too inconvenient for me, so sorry. It's just much easier to unpack a zip file (actually, it's really a great thing that actions on github just have archives, because on the main project's page none of such options are available and also it seems like there is no real need even of chocolatey for me anymore).
But thanks for suggesting anyway.

**Reply by CortexReaver** [reply]
Oh, now I realized what it is. This is windows 10 packages system or whatever it was called, it requires a windows appstore which I successfully disabled. There is some other way to install it via PowerShell, but this is all already too cryptic for me. I need to find right commands on google and pray to all gods in observed universe for not getting any errors (such as security errors related to account permissions).

**Reply by riverar** [reply]
Well, since you decided all the easy methods were inconvenient, you can also crack open the `.appinstaller` and harvest the package URL. So basically download the ZIP at https://install.eartrumpet.app/dev/EarTrumpet.Package_2.1.10.7_x86.appxbundle. Extract `EarTrumpet.Package_2.1.10.7_x86.appxbundle\EarTrumpet.Package_2.1.10.7_x86.appx\EarTrumpet` somewhere and you should be good to go. Be aware you'll miss out on the frequent updates in the dev branch.

**Reply by riverar** [reply]
I recognize we don't have a GitHub release feed of latest binaries, but the reality is users won't return and get important bug fixes if we do that. And we do not want to roll our own update mechanism in the app itself.

**Comment by CortexReaver** [comment]
I've noticed another thing. There should be an option to synchronize volume levels between all the devices, because after decreasing volume on one of them by a slider it stays desynced even if I use media keys. Yes, this is not a bug and the program is working as intended, but such an option will be a good UIX improvement.
Also have no intention to create a new request topic for this one, because it feels like all of this falls into the same category. Consider this as a workaround if the subject of this topic is impossible to implement with a working Windows 10 on-screen volume indicator.

**Reply by riverar** [reply]
Each device has its own volume level. Are you asking for a hotkey to set the levels for all devices to the same value? If so, that can be set up in Actions, for example:

![image](https://github.com/File-New-Project/EarTrumpet/assets/475132/eac85e0c-3167-40e6-a4ea-da41b57b7cb4)

**Reply by CortexReaver** [reply]
No, I mean when I drag one slider with the mouse button (or scroll wheel) itself it should be synched with the other ones. Like they are increasing and decreasing with the same amount as one that you modified. And such behavior should be optional (because of possibilities of a different setups) - a radiobutton option in settings.

**Reply by riverar** [reply]
Oh I see, like the Windows Volume Mixer (sndvol) does. It sounds like you want each slider to be relative to the master, I think. If so, we found that to be super confusing for users and decided against it. It's just not how mixers work in real life (pictured below). And in the new Windows 10 and Windows 11 volume mixers, it seems Microsoft abandoned this approach as well. I'm not opposed to the addition of a community option though.

![image](https://github.com/File-New-Project/EarTrumpet/assets/475132/ea8f16d6-32b5-4667-80bb-c03e13e57cb4)

**Reply by CortexReaver** [reply]
Relative to the master is one of the possible solutions. And, well, I mean if this will be optional it's totally fine, I guess... As I said earlier, the best solution will be the subject (not binding media keys to both sliders, but to one that is not default) with a working windows volume indicator.

**Reply by CortexReaver** [reply]
What I originally mean, and I think this is better:
right now the sliders moving on their own:

![28-06-2023 142152](https://github.com/File-New-Project/EarTrumpet/assets/34350690/c62c3d6b-9f35-4041-8671-a56512494369)

And what I propose is the option to sync them (red line is just for indication):

![28-06-2023 142438](https://github.com/File-New-Project/EarTrumpet/assets/34350690/93f7cc4f-1480-4db8-8ace-2fe4936dbfec)

If you move one of them, no matter which exactly, the others will be sliding into the exact same position. Same principle should be applied for the media keys (if such option was enabled).

**Reply by riverar** [reply]
Ah got it now thanks!
